lib/benchmarks/src/Cardano/Wallet/Benchmarks/Latency/BenchM.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Wallet.Benchmarks.Latency.BenchM where
+
+import Prelude
+
+import Cardano.Wallet.Api.Clients.Testnet.Shelley
+    ( A
+    )
+import Cardano.Wallet.Api.Types
+    ( ApiWallet
+    )
+import Cardano.Wallet.Benchmarks.Latency.Measure
+    ( LogCaptureFunc
+    )
+import Control.Monad
+    ( void
+    )
+import Control.Monad.Reader
+    ( MonadIO (..)
+    , MonadReader (..)
+    , ReaderT (..)
+    , ask
+    )
+import Control.Monad.Trans.Resource
+    ( MonadResource (..)
+    , ResourceT
+    , allocate
+    , runResourceT
+    )
+import Data.Generics.Internal.VL
+    ( (^.)
+    )
+import Data.Text.Class
+    ( FromText (..)
+    )
+import Network.Wai.Middleware.Logging
+    ( ApiLog
+    )
+import Numeric.Natural
+    ( Natural
+    )
+import Servant.Client
+    ( ClientEnv
+    , ClientError
+    , ClientM
+    , mkClientEnv
+    , parseBaseUrl
+    , runClientM
+    )
+import Test.Integration.Framework.DSL
+    ( Context
+    )
+
+import qualified Cardano.Wallet.Api.Clients.Testnet.Shelley as C
+import qualified Data.Text as T
+import qualified Test.Integration.Framework.DSL as DSL
+
+data BenchCtx = BenchCtx
+    { dslContext :: Context
+    , logFun :: LogCaptureFunc ApiLog ()
+    }
+
+-- one day we will export the manager from the context
+clientEnv :: Context -> ClientEnv
+clientEnv ctx = case parseBaseUrl $ show (fst $ ctx ^. #_manager) of
+    Left _ -> error "Invalid base URL"
+    Right bu -> mkClientEnv (snd $ ctx ^. #_manager) bu
+
+bombLeft :: Show l => Either l r -> r
+bombLeft = either (error . show) Prelude.id
```

Comment:
```suggestion
partialFromRight :: Show l => Either l r -> r
partialFromRight = either (error . show) Prelude.id
```

🤔 Perhaps this name is better?

* I like names that tell us about the success case, not about the failure case.
* The traditional prefix is `unsafe`. I always thought that it's a bit too harsh; `partial` could be better.